spec/factories/teacher_factory.rb

```diff
@@ -4,6 +4,11 @@
     sequence(:trs_first_name) { |n| "First name #{n}" }
     sequence(:trs_last_name) { |n| "Last name #{n}" }
 
+    trait :with_payments_frozen_years do
+      ect_payments_frozen_year { FactoryBot.create(:contract_period, year: [2021, 2022].sample).year }
+      mentor_payments_frozen_year { FactoryBot.create(:contract_period, year: [2021, 2022].sample).year }
```

Comment:
can we only set those values if there is an ect/mentor at school period present?